datafusion/expr/src/type_coercion/aggregates.rs

```diff
@@ -262,7 +243,7 @@ pub fn correlation_return_type(arg_type: &DataType) -> Result<DataType> {
 }
 
 /// function return type of an average
-pub fn avg_return_type(arg_type: &DataType) -> Result<DataType> {
+pub fn avg_return_type(func_name: &str, arg_type: &DataType) -> Result<DataType> {
```

Comment:
I wonder if this function is still needed or if we could put it all into AvgUDF (as a follow on PR)